src/game.js

```diff
@@ -1117,12 +1117,12 @@ export class Game {
 
 	findCreature(o) {
 		let ret = [],
-			o = $j.extend({
+			o2 = $j.extend({
```

Comment:
Was this related to a bug? I could be wrong but I suspect this was intentional so that all creature objects always had a type and team property... Meaning once it has been extended, it no-ops where as now it will process the extend every call.

Reply:
You can't actually do that in javascript because of hoisting. 

```javascript
// This
function a(o) {
  let o = jQuery.extends({
    foo: "bar"
  }, o)
}

// Is the same as this because of hoisting
function b(o) {
  let o; // O = undefined now
  o = jQuery.extends({
    foo: "bar"
  }, o)
}

// so the last one does this
o = jQuery.extends({
  foo: "bar"
}, undefined)
```

Reply:
These are 2 different examples though...

In one we are modifying the original argument (which JS does allow) and would propagate to the original object so long as it wasn't cloned...

The other you are redefining the variable...

This is not a case of hoisting.

Reply:
We are redefining it tough. Since after let ret = [] there isn't a semicolin but a comma. Again:
```javascript
let a = 5, b = 8
// Sugar for
let a = 5;
let b = 8;
```

Reply:
Ahh good catch, that was likely my doing when I cleaned things up the first time. I would hazard a guess it was not originally redefined, in its current form with the redefinition your explanation does make sense.